src/pkcs10.rs

```diff
@@ -113,8 +113,8 @@ impl<'e, K: DerWrite> DerWrite for CertificationRequestInfo<'e, K> {
     }
 }
 
-impl<K: BERDecodable> BERDecodable for CertificationRequestInfo<'static, K> {
-    fn decode_ber<'a, 'b>(reader: BERReader<'a, 'b>) -> ASN1Result<Self> {
+impl<'a, K: BERDecodable> BERDecodable for CertificationRequestInfo<'a, K> {
```

Comment:
Is this change compatible? There's a lifetime change here.

Reply:
Should be compatible. The original version implements `BERDecodable` only for `CertificationRequestInfo<'static, K>`. So any time you deserialize one of these structs, you would get one that has the static lifetime. The new version implements `BERDecodable` for *any* lifetime, which includes the static lifetime. 

You would think that implementing `BERDecodable` for the static lifetime should be sufficient to cover all cases, since it can always still be coerced to a shorter lifetime. I recently encountered a situation in which that wasn't the case, though. I don't completely understand why, but the compiler was giving me this error in a piece of async code:
```
error: implementation of `SomeTrait` is not general enough
   --> [...]
    | [code omitted]
    = note: `SomeStruct` must implement `SomeTrait<'0>`, for any lifetime `'0`...
    = note: ...but it actually implements `SomeTrait<'static>`
```
The `SomeTrait` was indeed only implemented for the static lifetime on `SomeStruct`, just like we originally had here for `CertificationRequestInfo`. Changing the trait implementation to cover all lifetimes solved that issue.